k8s/pkg/condition/lke.go

```diff
@@ -0,0 +1,43 @@
+package condition
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/linode/linodego"
+	"github.com/linode/linodego/k8s"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterHasReadyNode is a ClusterConditionFunc which polls for at least one node to have the
+// condition NodeReady=True.
+func ClusterHasReadyNode(ctx context.Context, options linodego.ClusterConditionOptions) (bool, error) {
+	clientset, err := k8s.BuildClientsetFromConfig(options.LKEClusterKubeconfig, options.TransportWrapper)
+	if err != nil {
+		return false, err
+	}
+
+	nodes, err := clientset.CoreV1().Nodes().List(ctx, v1.ListOptions{})
+	if err != nil {
+		return false, fmt.Errorf("failed to get nodes for cluster: %s", err)
+	}
+
+	for _, node := range nodes.Items {
+		for _, condition := range node.Status.Conditions {
+			if condition.Type == corev1.NodeReady && condition.Status == corev1.ConditionTrue {
+				return true, nil
+			}
+		}
+	}
+
+	return false, errors.New("no nodes in cluster are ready")
```

Comment:
Does it make sense to return an error if no nodes are ready? In general, I would expect a function like this to return `false` if the condition successfully evaluated to false, then for the upstream function using it to fail eventually on timeout.

Reply:
You make a point. I'll defer to @Charliekenney23 , but this PR is mostly just a refactor without functional change, so should probably be part of different PR

Reply:
John is right, it definitely doesn't match the k8s waiter convention. But yes, this is old code I added initially, so I can do this in a subsequent PR before we release.